### DISCUSSION TITLE
make got error with: ModuleNotFoundError: No module named 'setuptools'

### DISCUSSION THREAD

**Discussion by wwyl2000** [espnet/espnet#4492] [Q&A]
Hi, 
I am new with espnet.

When run:
make TH_VERSION=1.10.1 CUDA_VERSION=11.3

CUDA_VERSION=11.3
PYTHON=/usr/bin/python3
PYTHON_VERSION=Python 3.6.9
USE_CONDA=
TH_VERSION=1.10.1
. ./activate_python.sh && python3 -m pip install -e "..[train, recipe]"  # Install editable mode by default
Defaulting to user installation because normal site-packages is not writeable
Obtaining file:///home/xyz/work/espnet
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/xyz/work/espnet/setup.py'"'"'; __file__='"'"'/home/xyz/work/espnet/setup.py'"'"';f tr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'lose();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-3e50qh5q
       cwd: /home/xyz/work/espnet/
  Complete output (3 lines):
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
  ModuleNotFoundError: No module named 'setuptools'
  ----------------------------------------
WARNING: Discarding file:///home/xyz/work/espnet. Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
Makefile:105: recipe for target 'espnet.done' failed
make: *** [espnet.done] Error 1

I have python (3.7) and python3 (3.6.9) installed. In both, i can: import setuptools.  
Please help why i got the error
Thanks,
wwy

**Comment by kamo-naoyuki** [comment]
I don't recommend using the python installed by OS, anyway.  If you have some specific reason to use it, please continue to read.


Did you follow our documentation? Please show  your `activate_python.sh`.

I'm not sure why you don't have setuptools, but maybe you specified a wrong python path.
To install it, please type `. ./activate_python.sh && python3 -m pip install setuptools`, not using `pip install`, `pip3 install`

**Comment by wwyl2000** [comment]
Thanks for your reply.

I followed your doc: https://espnet.github.io/espnet/installation.html 
After git clone, i ran: 
Step1.  ./setup_anaconda.sh anaconda espnet 3.6.9
Step2. ./setup_venv.sh $(command -v python3)
Step3. ./setup_.sh $(command -v python3)
Step4. make TH_VERSION=1.10.1 CUDA_VERSION=11.3

but i got the problem. 
After failed many times, i tried: after above Step3, i ran: 
rm -f activate_python.sh && touch activate_python.sh
then Step4. Now i do not have the problem, but, the installation takes quite long time, as it installed several versions of the same module.

At Step1, i used python 3.6.9. Not sure if i am on the right track. not correct.

Thanks,
wwy

**Comment by wwyl2000** [comment]
If i do not run:
rm -f activate_python.sh && touch activate_python.sh

The content of activate_python.sh is as follows:

> #!/usr/bin/env bash
> # THIS FILE IS GENERATED BY tools/setup_python.sh
> export PYTHONUSERBASE="/home/xyz/work/espnet/tools/python_user_base"
> export PATH="/home/xyz/work/espnet/tools/python_user_base/bin":${PATH}
> export PATH=/usr/bin:${PATH}
>

**Comment by kamo-naoyuki** [comment]
Python3.6.9 is no longer supported.

**Comment by wwyl2000** [comment]
Thanks. I changed to python3.8, and run: 
./setup_anaconda.sh anaconda espnet 3.8

I still have: ~/work/espnet/tools/venv/lib/python3.6/ 
Why the directory name is python3.6/ not python3.8/  ?
Is this correct?

Thanks
wwy

**Comment by wwyl2000** [comment]
Run: ~/work/espnet/tools$ make TH_VERSION=1.10.1 CUDA_VERSION=11.3
I got the following:
CUDA_VERSION=11.3
PYTHON=/usr/bin/python3
PYTHON_VERSION=Python 3.6.9
USE_CONDA=
TH_VERSION=1.10.1
...

I can see: PYTHON_VERSION=Python 3.6.9, why it is still python 3.6.9, not 3.8?
